src/main/scala/machines/Internal.scala

```diff
@@ -2,5 +2,52 @@ package machines
 
 import regex._
 import dfa._
+import machines._
 
-// TODO: Add your code below
+// implicit conversion from char to regular language
+given Conversion[Char, RegularLanguage] = Character(_)
+
+
+// implicit conversion from string to regular language
+given Conversion[String, RegularLanguage] with
+    def apply(s: String): RegularLanguage = {
+        if (s.isEmpty)
+            Epsilon
+        else 
+            Concat(Character(s.head), apply(s.tail))
+    }
+
+// implicit conversion from regular language to DFA
+given Conversion[RegularLanguage, DFA] with
+    def apply(r: RegularLanguage): DFA = {
+        val alphabet = chars(r)
+        r.toDFA(using alphabet)
+    }
+
+// operators
+extension(l1: RegularLanguage) {
+    def ||(l2: RegularLanguage): RegularLanguage = Union(l1, l2)
+    def ~(l2: RegularLanguage): RegularLanguage = Concat(l1, l2)
+    def <*> = Star(l1)
+    def <+> = Concat(l1, Star(l1))
+    def apply(n: Int): RegularLanguage = {
+        if (n == 1)
+            l1
+        else
+            Concat(l1, apply(n-1))
+    }
+
+    // to DFA
+    def toDFA(using alphabet: Set[Char]): DFA = regexToDFA(l1, alphabet)
```

Comment:
Its pretty interesting that using this method requires us to explicitly define a default output for one of scala's own base classes (A set of chars), rather than one we created, and it makes me wonder if two programs both relying on different defaults for something like this could run into issues